### PR TITLE
Duration Presenter

### PR DESCRIPTION
**Pull request by felipeparaujo**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by votemike** [comment]
Does Cake not help with this? With it's `ChronosInterval` class or `diffForHumans()` methods?

File: src/DsAmen/Molecule/Duration/DurationPresenter.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    /** @var TranslateProvider */
+    private $translateProvider;
+
+    /**
+     * @var array
+     */
+    protected $options = [
+        'hTag' => 'h4',
+        'hTitle' => 'duration',
+        'show_icon' => true,
```
Comment (votemike) [inline src/DsAmen/Molecule/Duration/DurationPresenter.php]: Why is this snake_case but `showHeading` is camelCase?

File: src/DsAmen/Molecule/Duration/DurationPresenter.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        if ($this->getSeconds()) {
+            $ariaLabel .= ' ' . $tr->translate('time_seconds_long', ['%1' => $this->getSeconds()]);
+        }
+
+        return $ariaLabel;
+    }
+
+    public function getFormattedDuration()
```
Comment (votemike) [inline src/DsAmen/Molecule/Duration/DurationPresenter.php]: Typehints for all return types in this class?

File: src/DsAmen/Molecule/Duration/DurationPresenter.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        return $this->seconds;
+    }
+
+    public function getOptions()
+    {
+        return $this->options;
+    }
+
+    private function makeParts($seconds)
```
Comment (votemike) [inline src/DsAmen/Molecule/Duration/DurationPresenter.php]: Typehint $seconds?

**Comment by felipeparaujo** [comment]
It's not worth it using Chronos as we get the duration in seconds from the database. This is more efficient.

**Pull request merged**